core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1992,4 +1992,32 @@ package object config {
       .version("3.1.0")
       .doubleConf
       .createWithDefault(5)
+
+  private[spark] val SHUFFLE_NUM_PUSH_THREADS =
+    ConfigBuilder("spark.shuffle.push.numPushThreads")
+      .doc("Specify the number of threads in the block pusher pool. These threads assist " +
+        "in creating connections and pushing blocks to remote shuffle services when push based " +
+        "shuffle is enabled. By default, the threadpool size is equal to the number of cores.")
```

Comment:
Is it true or okay in YARN environment?
> By default, the threadpool size is equal to the number of cores.

Reply:
The number of cores here is referring to Spark executor cores, not the underlying CPU vcores.
Might need better clarification here.